src/decode-icmpv4.c

```diff
@@ -296,6 +313,20 @@ int DecodeICMPV4(ThreadVars *tv, DecodeThreadVars *dtv, Packet *p, const uint8_t
             }
             break;
 
+        case ICMP_ROUTERADVERT: {
+            /* packet points to beginning of icmp header */
+            ICMPV4RtrAdvert *icmpv4_router_advert = (ICMPV4RtrAdvert *)(pkt + sizeof(ICMPV4Hdr));
```

Comment:
Is not there a check missing like `if (p->payload_len < sizeof(ICMPV4RtrAdvert)) {` ?

Reply:
Ok, no check missing, but misleading comment

Reply:
Will adjust comment.